open_spiel/python/algorithms/efr.py

```diff
@@ -0,0 +1,967 @@
+# Copyright 2019 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Modified: 2023 James Flynn
+# Original: https://github.com/deepmind/open_spiel/blob/master/open_spiel/python/algorithms/cfr.py
+"""Python implementation of the extensive-form regret minimization algorithm.
+
+One iteration of EFR consists of:
+1) Compute current strategy from regrets (e.g. using Regret Matching).
+2) Compute values using the current strategy
+3) Compute regrets from these values
+
+The average policy converges to a Nash Equilibrium rather than the current policy as in CFR.
+"""
+import copy
+from collections import defaultdict
+import attr
+
+import numpy as np
+from scipy.linalg import lstsq
+
+import pyspiel
+from open_spiel.python import policy
+
+
+@attr.s
+class _InfoStateNode(object):
+    """An object wrapping values associated to an information state."""
```

Comment:
Throughout OpenSpiel, we use two spaces for indentation. Can you fix this and the other file to make it consistent?

Also can you run pylint with Google style files? (See step 9 of Adding a New Game here: https://github.com/google-deepmind/open_spiel/blob/master/docs/developer_guide.md#adding-a-game)

Reply:
Thanks for the comments!

I ran both files through pylint with the style guide and added a few more comments.

Let me know what you think, I am more than happy to make changes/clarify anything as required.